zebra/zebra_rnh.c

```diff
@@ -1344,13 +1344,17 @@ static void print_rnh(struct route_node *rn, struct vty *vty, json_object *json)
 	}
 
 	if (rnh->state) {
-		if (json)
+		if (json) {
 			json_object_string_add(
 				json_nht, "resolvedProtocol",
 				zebra_route_string(rnh->state->type));
-		else
+			json_object_string_addf(json_nht, "prefix", "%pFX",
+						&rnh->resolved_route);
+		} else {
 			vty_out(vty, " resolved via %s\n",
 				zebra_route_string(rnh->state->type));
+			vty_out(vty, " prefix %pFX\n", &rnh->resolved_route);
```

Comment:
IMO make the line “ resolved via %s prefix %pFX\n”.

Reply:
Ok, I have also added a "," between the protocol and the prefix.